src/backend/main/go.main/cmd/caliopen/cli-cmds/APIs.go

```diff
@@ -0,0 +1,206 @@
+	"os/signal"
+	"syscall"
+)
+
+var (
+	withSmtp    bool
+	withPyramid bool
+	withTcp     bool
+
+	APIsCmd = &cobra.Command{
+		Use:   "APIs",
+		Short: "start the caliopen application's APIs servers",
+		Run:   APIs,
+	}
+
+	signalChannel = make(chan os.Signal, 1) // for trapping SIG_HUP
+)
+
+func init() {
+	APIsCmd.PersistentFlags().BoolVar(&withSmtp, "smtp", true, "Start the embedded smtp server")
+	APIsCmd.PersistentFlags().BoolVar(&withPyramid, "pyramid", true, "Launch the python/pyramid API (pserve script)")
+	APIsCmd.PersistentFlags().BoolVar(&withTcp, "tcp", true, "Start the REST TCP API")
+	RootCmd.AddCommand(APIsCmd)
+}
+
+func sigHandler() {
+	// handle SIGHUP for reloading the configuration while running
+	signal.Notify(signalChannel, syscall.SIGHUP, syscall.SIGTERM, syscall.SIGQUIT, syscall.SIGINT, syscall.SIGKILL)
+
+	for sig := range signalChannel {
+
+		if sig == syscall.SIGHUP {
+			err := readConfig(false)
+			if err != nil {
+				log.WithError(err).Error("Error while ReadConfig (reload)")
+			} else {
+				log.Infof("Configuration is reloaded at %s", guerrilla.ConfigLoadTime)
+			}
+			// TODO: reinitialize
+		} else if sig == syscall.SIGTERM || sig == syscall.SIGQUIT || sig == syscall.SIGINT {
+			log.Info("Shutdown signal caught")
+
+			if withPyramid {
+				//pyramid shutdown
+				log.Info("…shutdowning pymarid")
+				pyPID, err := afero.ReadFile(AppsFS, "pyramid.pid")
+				if err != nil {
+					log.WithError(err).Info("unable to read pyramid.pid file. You may have to kill pserve process.")
+				}
+				exec.Command("kill", string(pyPID)).Run()
+			}
+
+			if withTcp {
+
+			}
+
+			if withSmtp {
+				log.Info("…shutdowning smtp")
+			}
+			//app.Shutdown()
+			log.Infof("Shutdown completed, exiting.")
+			os.Exit(0)
+		} else {
+			os.Exit(0)
+		}
+	}
+}
+
+func APIs(cmd *cobra.Command, args []string) {
+	err := readConfig(false)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// start HTTP reverse proxy
+	go rest_api.StartProxy()
+
+	// start GO REST server
+	//load API config
+	v := viper.New()
+	v.SetConfigName(cmdConfig.Apis.GoRESTconfigFile) // name of config file (without extension)
+	v.AddConfigPath(cmdConfig.Apis.GoRESTconfigPath) // path to look for the config file in
+	v.AddConfigPath(configPath)
+	v.AddConfigPath("$CALIOPENROOT/src/backend/configs/") // call multiple times to add many search paths
+	v.AddConfigPath(".")                                  // optionally look for config in the working directory
+
+	err = v.ReadInConfig() // Find and read the config file*/
+	if err != nil {
+		log.WithError(err).Fatalf("Could not read main config file <%s>.", cmdConfig.Apis.GoRESTconfigFile)
+	}
+	apiConfig := rest_api.APIConfig{}
+	err = v.Unmarshal(&apiConfig)
+	if err != nil {
+		log.WithError(err).Fatalf("Could not parse config file: <%s>", cmdConfig.Apis.GoRESTconfigFile)
+	}
+
+	err = rest_api.InitializeServer(apiConfig)
+	if err != nil {
+		log.Fatal(err)
+	}
+	go rest_api.StartServer()
+
+	if withPyramid {
```

Comment:
I'll remove that bootstraping, but notice that go.main is not an « http server ».  
In my mind, it is the « master of ceremony », ie the binary that manages the whole Caliopen's instance : reads the configuration file and exports the settings to other processes, start services/processes, monitor them (keeps all the other processes running. If a child process dies, another one is restarted automatically),  is the centralized logging facility, etc…  
But let's forget it for now.